src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChainBuilder.java

```diff
@@ -98,12 +99,14 @@ public ItineraryListFilterChainBuilder addGroupBySimilarity(GroupBySimilarity gr
   }
 
   /**
-   * This function is used to compute a max-limit for generalized-cost. The limit is applied to
-   * itineraries with at least one transit leg. Street-only itineraries are not considered.
+   * This function is used to compute a max-limit for generalized-cost, together with
+   * waitAtBeginningOrEndCostFactor. The limit is applied to itineraries with at least one transit
+   * leg. Street-only itineraries are not considered.
    * <p>
-   * The smallest transit leg generalized-cost value is used as input to the function. For example
-   * if the function is {@code f(x) = 1800 + 2.0 x} and the smallest cost is {@code 5000}, then all
-   * transit itineraries with a cost larger than {@code 1800 + 2 * 5000 = 11 800} is dropped.
+   * All itineraries are compared to each other using this function. For example if the function is
+   * {@code f(x) = 1800 + 2.0 x} and one of the itineraries has a cost of {@code 5000}, then all
+   * transit itineraries with a cost larger than {@code 1800 + 2 * 5000 = 11 800} plus the cost for
+   * wait time multiplied by waitAtBeginningOrEndCostFactor are dropped.
```

Comment:
I think we can merge this with the `withWaitAtBeginningOrEndCostFactor` method. To apply the `TransitGeneralizedCostLimit` you need to provide all 3 parameters - theJavaDoc also become easier to read if all params are in one place. Not sure if we should drop the `DoubleFunction` and just use 3 parameters, but that might be the cleanest.
```
  /**
   * The given parameters is used to compare all itineraries with each other, droping itineraries 
   * with a high relative cost. The limit is applied to itineraries with at least one transit
   * leg. Street-only itineraries are not considered.
   * <p>
   * For all pairs {code (i1, i2)} in the set of returned itineraries, a generalized cost limit is
   * computed. If the generalized-cost of i1 is less then the generalized-cost of i2, i2 is dropped if:
   * <pre>
   * t0 := is the time between i1 and i2.
   * 
   * limit = i1.generalized-cost * costFactor + minimumCostDifference - t0 * itineraryIntervallRelaxFactor
   *
   * i2 is dropped if i2.generalized-cost is greater than the limit. 
   * </pre>
   * @param minimumCostDifference the minimum cost diffrence between two itineraries. Unit: "transit second"
   * @param costFactor            the facor multiplied to the smallest cost to get the limit. 
   * @param intervallRelaxFactor  the factor multiplied with the time between to itineraries to prevent an
   *                              itinerary to dominate another when the time between them is large.  
   */
  public ItineraryListFilterChainBuilder withTransitGeneralizedCostLimit(
     	int minimumCostDifference,
     	double costFactor,
        double intervalRelaxFactor
  ) {
```